src/cmd/commit.ts

```diff
@@ -32,52 +42,82 @@ export const gitPush = async (branch: string): Promise<boolean> => {
     d.info(e.stdout)
     d.error(e.stderr)
     return false
-  } finally {
-    cd(rootDir)
   }
 }
 
-export const commit = async (): Promise<void> => {
-  const d = terminal('commit')
-  await validateValues()
-  d.info('Preparing values')
-  const values = await hfValues()
-  cd(env.ENV_DIR)
-  await preCommit()
-  await encrypt()
-  d.info('Committing values')
-  cd(env.ENV_DIR)
+const setDeplymentStatus = async (): Promise<void> => {
+  const status = await getOtomiDeploymentStatus()
+  if (status !== 'deployed') {
+    await nothrow($`kubectl -n ${otomiStatusNamespace} create cm ${otomiStatusCmName} --from-literal=status='deployed'`)
+    // Since status is an indicator of successful deployment, the generated passwords must be deleted later.
+    await nothrow($`kubectl delete secret ${otomiPasswordsSecretName}`)
+  }
+}
+
+const setEnv = (values: any): void => {
+  const isCertStaging = values.charts?.['cert-manager']?.stage === 'staging'
+  if (isCertStaging) {
+    process.env.GIT_SSL_NO_VERIFY = 'true'
+    process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0'
+  }
+}
+const waitForGitea = async (values: any): Promise<void> => {
+  const giteaEnabled = values?.charts?.gitea?.enabled ?? true
+  if (giteaEnabled && (!env.CI || isChart)) {
+    const healthUrl = (await $`git config --get remote.origin.url`).stdout.trim()
+    debug.debug('healthUrl: ', healthUrl)
+    await waitTillAvailable(healthUrl)
+  }
+}
+
+const getGitBranch = (values: any): string => {
+  let branch = 'main'
+  if (values.charts?.gitea?.enabled === false) {
+    branch = values.charts!['otomi-api']!.git!.branch ?? branch
+  }
+  return branch
+}
+
+const commitAndPush = async (branch, pullBeforePush = false): Promise<void> => {
+  const d = terminal('commitAndPush')
   await $`git add -A`
   try {
     await $`git commit -m 'otomi commit' --no-verify`
   } catch (e) {
     d.info(e.stdout)
-    d.error(e.stderr)
+    d.error(e)
```

Comment:
since stdout was just printed above, can we just print the error here? it was fine before

Reply:
Not every command prints output on failure, instead return exit code.
Below example presents difference between two ways of printing error.
For comparison:
```
console.log(e.stderr)
ls: cannot access 'dadsad': No such file or directory

console.log(e)
ProcessOutput {
  stdout: '',
  stderr: "ls: cannot access 'dadsad': No such file or directory\n",
  exitCode: 2 (Misuse of shell builtins)
}

````